server/game/cards/01_SOR/leaders/EmperorPalpatineGalacticRuler.ts

```diff
@@ -0,0 +1,69 @@
+import AbilityHelper from '../../../AbilityHelper';
+import { LeaderUnitCard } from '../../../core/card/LeaderUnitCard';
+import { RelativePlayer, WildcardCardType } from '../../../core/Constants';
+
+export default class EmperorPalpatineGalacticRuler extends LeaderUnitCard {
+    protected override getImplementationId() {
+        return {
+            id: '5784497124',
+            internalName: 'emperor-palpatine#galactic-ruler',
+        };
+    }
+
+    protected override setupLeaderSideAbilities() {
+        this.addActionAbility({
+            title: 'Deal 1 damage to a unit and draw a card',
+            cost: [
+                AbilityHelper.costs.abilityResourceCost(1),
+                AbilityHelper.costs.exhaustSelf(),
+                AbilityHelper.costs.defeat({
+                    cardTypeFilter: WildcardCardType.Unit,
+                    controller: RelativePlayer.Self
+                })
+            ],
+            immediateEffect: AbilityHelper.immediateEffects.simultaneous([
+                AbilityHelper.immediateEffects.selectCard({
+                    cardTypeFilter: WildcardCardType.Unit,
+                    innerSystem: AbilityHelper.immediateEffects.damage({ amount: 1 })
+                }),
+                AbilityHelper.immediateEffects.draw((context) => ({ target: context.source.controller }))
+            ])
+        });
+    }
+
+    protected override setupLeaderUnitSideAbilities() {
+        this.addTriggeredAbility({
+            title: 'Take control of a damaged non-leader unit',
+            when: {
+                onLeaderDeployed: (event, context) => event.card === context.source
```

Comment:
Would it make sense for us to add a "this.addDeployAbility" helper method?

Reply:
I thought about but there's not that many cards it would apply to and there are other triggers with equal impact that we haven't added helpers for. I don't want to have a huge number of them